From 245bc3b53f385e12e4dc9d2cb765a55e10e0fdc5 Mon Sep 17 00:00:00 2001
From: progier389 <progier@redhat.com>
Date: Fri, 13 Feb 2026 17:51:12 +0100
Subject: [PATCH] Issue 3134 - Fix build break (#7260)

Fix build break of PR #7238 related to import rpm

Issue: #3134

Reviewed by: @vashirov (Thanks!)
---
 src/lib389/lib389/utils.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/lib389/lib389/utils.py b/src/lib389/lib389/utils.py
index 07cb34d93..4df36cf1f 100644
--- a/src/lib389/lib389/utils.py
+++ b/src/lib389/lib389/utils.py
@@ -28,7 +28,6 @@ from datetime import (datetime, timedelta)
 import sys
 import filecmp
 import pwd
-import rpm
 import shlex
 import operator
 import subprocess
@@ -2135,6 +2134,8 @@ def get_timeout_scale():
 
 def rpm_is_older(pkg, version):
     """Check if an RPM package version is older than specified version"""
+    # rpm module is not installed in build environment so let import it only when used.
+    import rpm
     ts = rpm.TransactionSet()
     mi = ts.dbMatch('name', pkg)
     for h in mi:
-- 
2.52.0

